.travis.yml

```diff
@@ -22,16 +21,17 @@ matrix:
           packages:
             - ksh
             - zsh
-      env:
-        - PATH=$HOME/bin/:$PATH
-        - SHFMT_VERSION=v2.6.2
-        - SHFMT_DOWNLOAD_URL="https://github.com/mvdan/sh/releases/download/${SHFMT_VERSION}/shfmt_${SHFMT_VERSION}_linux_amd64"
-      before_script:
-        - $SHELL --version 2>/dev/null || dpkg -s "$SHELL" 2>/dev/null || command -v "$SHELL"
-        - curl --version
-        - shellcheck --version
-        - curl -sSL "$SHFMT_DOWNLOAD_URL" -o $HOME/bin/shfmt;
-        - chmod +x $HOME/bin/shfmt;
-        - shfmt --version
+      script:
+        - ./install_test.sh
+
+    - language: bash
+      os:
+        - osx
+      addons:
+        homebrew:
+          packages:
+            - ksh
+            - zsh
```

Comment:
ksh and zsh are not needed.

Reply:
Why are they not needed?

Are they not supported?

Do they not need to be tested for some technical reason?

I don't understand.

Reply:
Okay, but running `sh` in CI on `linux` is not running `sh` but `dash`. And running `sh` in CI on `osx` is `bash`.

Is every script that is gonna run in `dash` gonna run fine in `zsh`, too? Is every script that is gonna run in `bash` gonna run fine in `zsh`, too?

Is there something I'm missing here? I'm kinda new to shell-scripting.

Reply:
"sh" is the lowest common denominator - if it works under that it'll work.

Reply:
@ry Would you approve this PR if I put the `zsh` and `ksh` tests into an if-CI block?

Reply:
> I know that `sh` is the lowest common denominator. But again, `/bin/sh` is symlinking to `/bin/bash` on Mac and to `/bin/dash` on Ubuntu. So in that sense there's no "if it works under `sh`, then it'll work everywhere". If you follow the symlinks, it only works under `bash` or `dash` because those are completely different shells. For example, `set -o pipefail` works in Bash, but not in Dash. That's why the CI was failing for me a few times. If we would have shipped that, we would have broken the installation for Ubuntu users or people using the script in Linux jobs on Travis. `shellcheck` did _not_ warn about `set -o pipefail` being a bashism so linting didn't help either.

For sake of technical correctness `/bin/sh` is not symlinked on mac. Instead you have dedicated _proxy_ binary that invokes `/bin/bash` with `--posix` flag thus limiting its feature set to emulate POSIX shell aka `sh`. As you noticed that downgrade is not perfect same way `dash` might not be equally restrictive as `sh` would be. Bottom line removing testing inside actual shells **is a mistake**.
Take [`nvm`](https://github.com/creationix/nvm) for example (shell script itself). Their installer is `bash` only so they test only in `bash` but all other parts are tested across wide variety of common shells user might use: https://github.com/creationix/nvm/blob/master/.travis.yml